ignite/pkg/cosmosclient/cosmosclient.go

```diff
@@ -251,92 +309,65 @@ func (r Response) Decode(message proto.Message) error {
 }
 
-// ConsensusInfo returns the appropriate tendermint consensus state by given height
-// and the validator set for the next height
-func (c Client) ConsensusInfo(ctx context.Context, height int64) (ConsensusInfo, error) {
-	node, err := c.Context().GetNode()
+// BroadcastTx creates and broadcasts a tx with given messages for account.
+func (c Client) BroadcastTx(accountName string, msgs ...sdktypes.Msg) (Response, error) {
+	_, broadcast, err := c.BroadcastTxWithProvision(accountName, msgs...)
 	if err != nil {
-		return ConsensusInfo{}, err
+		return Response{}, err
 	}
+	return broadcast()
+}
 
-	commit, err := node.Commit(ctx, &height)
+func (c Client) GenerateTx(accountName string, msgs ...sdktypes.Msg) (string, error) {
+	unsignedTx, _, err := c.BroadcastTxWithProvision(accountName, msgs...)
 	if err != nil {
-		return ConsensusInfo{}, err
+		return "", err
 	}
 
-	var (
-		page  = 1
-		count = 10_000
-	)
-	validators, err := node.Validators(ctx, &height, &page, &count)
+	json, err := c.context.TxConfig.TxJSONEncoder()(unsignedTx.GetTx())
 	if err != nil {
-		return ConsensusInfo{}, err
+		return "", err
 	}
 
-	protoValset, err := tmtypes.NewValidatorSet(validators.Validators).ToProto()
-	if err != nil {
-		return ConsensusInfo{}, err
-	}
+	return string(json), nil
+}
 
-	heightNext := height + 1
-	validatorsNext, err := node.Validators(ctx, &heightNext, &page, &count)
-	if err != nil {
-		return ConsensusInfo{}, err
-	}
+// protects sdktypes.Config.
+var mconf sync.Mutex
 
-	var (
-		hash = tmtypes.NewValidatorSet(validatorsNext.Validators).Hash()
-		root = commitmenttypes.NewMerkleRoot(commit.AppHash)
-	)
+func (c Client) lockBech32Prefix() (unlockFn func()) {
+	mconf.Lock()
+	config := sdktypes.GetConfig()
+	config.SetBech32PrefixForAccount(c.addressPrefix, c.addressPrefix+"pub")
 
-	return ConsensusInfo{
-		Timestamp:          commit.Time.Format(time.RFC3339Nano),
-		NextValidatorsHash: bytes.HexBytes(hash).String(),
-		Root:               base64.StdEncoding.EncodeToString(root.Hash),
-		ValidatorSet:       protoValset,
-	}, nil
+	return mconf.Unlock
 }
 
-// Status returns the node status
-func (c Client) Status(ctx context.Context) (*ctypes.ResultStatus, error) {
-	return c.RPC.Status(ctx)
-}
+func performQuery[T any](c Client, q func() (T, error)) (T, error) {
+	unlockFn := c.lockBech32Prefix()
+	defer unlockFn()
 
-// BroadcastTx creates and broadcasts a tx with given messages for account.
-func (c Client) BroadcastTx(accountName string, msgs ...sdktypes.Msg) (Response, error) {
-	_, broadcast, err := c.BroadcastTxWithProvision(accountName, msgs...)
-	if err != nil {
-		return Response{}, err
-	}
-	return broadcast()
+	return q()
 }
 
-// protects sdktypes.Config.
-var mconf sync.Mutex
-
 func (c Client) BroadcastTxWithProvision(accountName string, msgs ...sdktypes.Msg) (
-	gas uint64, broadcast func() (Response, error), err error) {
+	unsignedTx client.TxBuilder, broadcast func() (Response, error), err error) {
```

Comment:
I changed for this to return the unsigned transaction instead so that this could be used for generate-only commands. You simply don't call the broadcast function.
You can still get the gas from the unsigned tx.